crates/extensions/c8y_auth_proxy/src/url.rs

```diff
@@ -0,0 +1,51 @@
+use miette::Context;
+use miette::IntoDiagnostic;
+use tedge_config::TEdgeConfig;
+
+pub struct ProxyUrlGenerator {
+    port: u16,
+}
+
+impl ProxyUrlGenerator {
+    pub fn new(port: u16) -> Self {
+        Self { port }
+    }
+
+    pub fn from_tedge_config(tedge_config: &TEdgeConfig) -> Self {
+        Self {
+            port: tedge_config.c8y.proxy.bind.port,
+        }
+    }
+
+    pub fn proxy_url(&self, cumulocity_url: &str) -> miette::Result<url::Url> {
+        let mut parsed_url = cumulocity_url
+            .parse::<url::Url>()
+            .into_diagnostic()
+            .wrap_err_with(|| format!("failed to parse url: {cumulocity_url}"))?;
+        parsed_url
+            .set_host(Some("localhost"))
```

Comment:
No, this must be an IP address accessible from child-devices. See comment on `pub struct ProxyUrlGenerator`.